src/Microsoft.OData.Client/XmlConstants.cs

```diff
@@ -982,6 +982,9 @@ internal static class XmlConstants
         /// <summary>edm string primitive type name</summary>
         internal const string EdmDateTimeOffsetTypeName = "Edm.DateTimeOffset";
 
+        /// <summary>edm string primitive type name</summary>
+        internal const string EdmDateTimeTypeName = "Edm.DateTime";

```

Comment:
Please delete this change if the second parameter of the line added in PrimitiveType.cs could be null.